torch/fft/__init__.py

```diff
@@ -9,18 +9,227 @@
 # connects the torch.fft Python namespace to the torch._C._fft builtins.
 
 fft = _add_docstr(_fft.fft_fft, r"""
-fft(input) -> Tensor
+fft(input: Tensor, n: Optional[int] =None, dim: int =-1, norm: Optional[str] =None) -> Tensor
```

Comment:
Despite the `-> Tensor` suggesting we type these strings (maybe we should?) we typically write them like normal Python signatures: `input, n=None, dim=-1, norm=None`

Reply:
Would a note explaining how fft(real-valued tensor) is distinct from rfft be helpful?

Reply:
Remove the type annotations.

Reply:
>Would a note explaining how fft(real-valued tensor) is distinct from rfft be helpful?

There is a note to that effect in the `rfft` docs. Do you think it belongs here as well?

Reply:
I noticed that. I would still include a note in fft, I think. Most people using the docs are probably just looking at one function returned by the search feature, so the note in rfft will be hard to discover for people looking at fft, and I think the interpretation by fft of non-complex dtype inputs as complex dtype inputs is not immediately obvious.

Reply:
Added a note about hermitian symmetry and `rfft` to the `fft` doc.